File: vscode/src/completions/tree-sitter/query-tests/test-data/intents.snap.ts
// 
// | - query start position in the source file.
// █ – query start position in the annotated file.
// ^ – characters matching the last query result.
//
// ------------------------------------

  function wrapper() {
      console.log('wrapper')
      function test() {
//                    ^ start function.body[1]
//                    █
      }
//    ^ end function.body[1]
  }

// Nodes types:
// function.body[1]: statement_block

// ------------------------------------

  function testParams() {
//                   ^^ function.parameters[1]
//                   █
      wrapper
  }

// Nodes types:
// function.parameters[1]: formal_parameters

// ------------------------------------

  function arrowWrapper() {
      const arrow = (value: string) => {
//                                     ^ start function.body[1]
//                                     █
      }
//    ^ end function.body[1]
  }

// Nodes types:
// function.body[1]: statement_block

// ------------------------------------

  function signature()
//                  ^^ function.parameters[1]
//                  █

// Nodes types:
// function.parameters[1]: formal_parameters

// ------------------------------------

// comment
//^^^^^^^^^^ comment[1]
//         █

// Nodes types:
// comment[1]: comment

// ------------------------------------

  function functionName() {}
//         ^^^^^^^^^^^^ function.name[1]
//                    █

// Nodes types:
// function.name[1]: identifier

// ------------------------------------

  const stringValue = "hello"
//                     ^^^^^ string[1]
//                       █

// Nodes types:
// string[1]: string_fragment

// ------------------------------------

  const templateListeralValue = `world`
//                              ^^^^^^^ string[1]
//                               █

// Nodes types:
// string[1]: template_string

// ------------------------------------

  function withEmptyBlockStatement() {
      functionName(); { }
//                    ^^^ block_statement[1]
//                    █
  }

// Nodes types:
// block_statement[1]: statement_block

// ------------------------------------

  function returnStatement() {
      return
//    ^^^^^^ return_statement[1]
//         █
  }

// Nodes types:
// return_statement[1]: return_statement

// ------------------------------------

  function returnStatementValue() {
      return "asd"
//           ^^^^^ string[1]
//           █
  }

// Nodes types:
// string[1]: string

// ------------------------------------

  interface TestInterface {
//                        ^ start type_declaration.body[1]
//                        █
  }
//^ end type_declaration.body[1]

// Nodes types:
// type_declaration.body[1]: object_type

// ------------------------------------

  type TestType = {
//                ^ start type_declaration.body[1]
//                █
  }
//^ end type_declaration.body[1]

// Nodes types:
// type_declaration.body[1]: object_type

File: vscode/src/completions/tree-sitter/query-tests/test-data/intents.snap.py
# 
# | - query start position in the source file.
# █ – query start position in the annotated file.
# ^ – characters matching the last query result.
#
# ------------------------------------

 def wrapper():
     print('wrapper')
     def test():
#              █
         pass
#        ^^^^ function.body[1]

# Nodes types:
# function.body[1]: block

# ------------------------------------

 def test_params():
#               ^^ function.parameters[1]
#               █
     wrapper()

# Nodes types:
# function.parameters[1]: parameters

# ------------------------------------

 def test_parameter(val):
#                   ^^^ parameter[1]
#                   █
     wrapper()

# Nodes types:
# parameter[1]: identifier

# ------------------------------------

 def arrow_wrapper():
     arrow = lambda value: None
#                          ^^^^ function.body[1]
#                        █

# Nodes types:
# function.body[1]: none

# ------------------------------------

 class Agent:
#           █
     pass
#    ^^^^ class.body[1]

# Nodes types:
# class.body[1]: block

# ------------------------------------

 def signature():
#               █
     pass
#    ^^^^ function.body[1]

# Nodes types:
# function.body[1]: block

# ------------------------------------

# comment
#^^^^^^^^^ comment[1]
#  █

# Nodes types:
# comment[1]: comment

# ------------------------------------

 def function_name():
#    ^^^^^^^^^^^^^ function.name[1]
#                █
     pass

# Nodes types:
# function.name[1]: identifier

# ------------------------------------

 def return_statement():
     return
#    ^^^^^^ return_statement[1]
#         █

# Nodes types:
# return_statement[1]: return_statement

# ------------------------------------

 def return_statement_value(value: str, flag: bool = False):
     return 'asd'
#           ^^^^^ return_statement.value[1]
#           █

# Nodes types:
# return_statement.value[1]: string

# ------------------------------------

 return_statement_value('value')
#                       ^^^^^^^ argument[1]
#                       █

# Nodes types:
# argument[1]: string

# ------------------------------------

 return_statement_value('value', False)
#                                ^^^^^ argument[1]
#                                █

# Nodes types:
# argument[1]: false

# ------------------------------------

 return_statement_value()
#                      ^^ arguments[1]
#                      █

# Nodes types:
# arguments[1]: argument_list

# ------------------------------------

 from math import cos
#     ^^^^ import.source[1]
#     █

# Nodes types:
# import.source[1]: dotted_name

# ------------------------------------

 from math import comb
#                 ^^^^ import.name[1]
#                 █

# Nodes types:
# import.name[1]: dotted_name

